Riot/Modules/Authentication/Views/AuthInputsView.m

```diff
@@ -611,13 +611,15 @@ - (void)prepareParameters:(void (^)(NSDictionary *parameters, NSError *error))ca
                         
                         // Launch email validation
                         submittedEmail = [[MXK3PID alloc] initWithMedium:kMX3PIDMediumEmail andAddress:self.emailTextField.text];
+                        
+                        NSString *identityServer = restClient.identityServer;
 
                         // Create the next link that is common to all Vector.im clients
                         NSString *nextLink = [NSString stringWithFormat:@"%@/#/register?client_secret=%@&hs_url=%@&is_url=%@&session_id=%@",
                                               [Tools webAppUrl],
                                               [submittedEmail.clientSecret stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLHostAllowedCharacterSet]],
                                               [restClient.homeserver stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLHostAllowedCharacterSet]],
-                                              [restClient.identityServer stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLHostAllowedCharacterSet]],
+                                              [identityServer stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLHostAllowedCharacterSet]],
```

Comment:
Caution: What should be done when this `identityServer ` string is null? remove "is_url" from the path?